src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest_dischargeToHospital.kt

```diff
@@ -0,0 +1,727 @@
+
+        bookingId = findBookingId(offenderType)
+
+        dataLoaderTransaction.get {
+          testDataContext.getMovements(bookingId!!)
+            .also {
+              with(it.last()) {
+                assertThat(fromAgency.id).isEqualTo("SYI")
+                assertThat(toAgency.id).isEqualTo("HAZLWD")
+                assertThat(movementDate).isEqualTo(LocalDate.now())
+                assertThat(isActive).isTrue()
+                assertThat(movementReason.code).isEqualTo(DISCHARGE_TO_PSY_HOSPITAL.code)
+                assertThat(commentText).contains("Psychiatric Hospital Discharge to Hazelwood House")
+              }
+            }
+        }
+      }
+
+      @ParameterizedTest
+      @MethodSource("uk.gov.justice.hmpps.prison.api.resource.impl.OffenderResourceIntTest_dischargeToHospital#getOffenderTypes")
+      fun `should make old movements inactive`(offenderType: OffenderType) {
+        createOffender(offenderType)
+
+        dischargeToHospital(offenderNo, dischargeRequest()).isOk
+
+        bookingId = findBookingId(offenderType)
+
+        testDataContext.getMovements(bookingId!!)
+          .filter { it.isActive }
+          .also { assertThat(it).hasSize(1) }
+      }
+
+      @ParameterizedTest
+      @MethodSource("uk.gov.justice.hmpps.prison.api.resource.impl.OffenderResourceIntTest_dischargeToHospital#getOffenderTypes")
+      fun `should create a case note for the release`(offenderType: OffenderType) {
+        createOffender(offenderType)
+
+        dischargeToHospital(offenderNo, dischargeRequest()).isOk
+
+        bookingId = findBookingId(offenderType)
+
+        testDataContext.getCaseNotes(offenderNo)
+          .maxBy { it.caseNoteId }
+          .also {
+            assertThat(it.type).isEqualTo("PRISON")
+            assertThat(it.subType).isEqualTo("RELEASE")
+            // TODO SDIT-549 Shouldn't the case note be updated if the offender was released before moving to hospital? This feels like a bug.
+            if (offenderType != RELEASED) {
+              assertThat(it.text).isEqualTo("Transferred from SHREWSBURY for reason: Moved to psychiatric hospital Hazelwood House.")
+            }
+            assertThat(it.creationDateTime.toLocalDate()).isEqualTo(LocalDate.now())
+            assertThat(it.agencyId).isEqualTo("SYI")
+          }
+      }
+
+      @ParameterizedTest
+      @MethodSource("uk.gov.justice.hmpps.prison.api.resource.impl.OffenderResourceIntTest_dischargeToHospital#getOffenderTypes")
+      fun `should update the bed assignment history`(offenderType: OffenderType) {
+        createOffender(offenderType)
+
+        // check bed assignments not ended (yet) - not relevant if no booking or released
+        if (listOf(BOOKING, TEMPORARY_ABSENCE).contains(offenderType)) {
+          testDataContext.getBedAssignments(bookingId!!)
+            .maxBy { it.assignmentDate }
+            .also {
+              assertThat(it.assignmentEndDate).isNull()
+            }
+        }
```

Comment:
We have a few of these were the pre-checks don't apply to some offender types... which isn't ideal but didn't want to duplicate the various tests if possible.